packages/adblocker-electron/adblocker.ts

```diff
@@ -7,17 +27,125 @@ export function fromElectronDetails({
   url,
   resourceType,
   referrer,
-}: {
-  url: string;
-  resourceType: ElectronRequestType;
-  referrer: string;
-}): Request {
+}: Electron.OnBeforeRequestDetails | Electron.OnHeadersReceivedDetails): Request {
   return Request.fromRawDetails({
     sourceUrl: referrer,
-    type: resourceType || 'other',
+    type: (resourceType || 'other') as ElectronRequestType,
     url,
   });
 }
 
+/**
+ * Wrap `FiltersEngine` into a Electron-friendly helper class.
+ */
+export class ElectronBlocker extends FiltersEngine {
+  public enableBlockingInSession(ses: Electron.Session) {
+    ses.webRequest.onHeadersReceived({ urls: ['<all_urls>'] }, this.onHeadersReceived);
+    ses.webRequest.onBeforeRequest({ urls: ['<all_urls>'] }, this.onBeforeRequest);
+    ipcMain.on('get-cosmetic-filters', this.onGetCosmeticFilters);
+    ses.setPreloads([join(__dirname, './content.js')]);
+  }
+
+  private onGetCosmeticFilters = (
+    e: Electron.IpcMessageEvent,
+    url: string,
+    id: string,
```

Comment:
Was thinking that we might not need `id` at all here (and same in `content.ts`). The flow would be:

1. preload script sends message via ipcRenderer to main process (channel `get-cosmetics-filters`)
2. main process receives the `event` + `url` + `msg` as arguments
3. main process returns answer using `event.reply(msg)` instead of `event.sender.send(...)`; but using a generic channel like `get-cosmetics-filters-response`.

What do you think? If that works it would simplify the communication logic a bit.

Reply:
Well, there are TypeScript errors with `event.reply`

Reply:
Yep, I think type definitions are not up-to-date, but it worked! Also, seems like this is going to become the recommended way from what I could read in issue tracker.